lib/dialects/mysql/query.js

```diff
@@ -48,9 +48,20 @@ Query.prototype.run = function(sql, parameters) {
 
         reject(self.formatError(err));
       } else {
-        resolve(self.formatResults(results));
+        resolve(results);
       }
     }).setMaxListeners(100);
+  })
+  // Log warnings if we've got them.
+  .then(function(results){
+    if (results && results.warningCount > 0) {
+      return self.logWarnings(results);
```

Comment:
Trying the recursive call. `Query.prototype.run()` doesn't take those arguments, but the recursive call seems to be okay using the same arguments. Since `SHOW WARNINGS` isn't really a query which could result in a warning itself, I'm not worried about infinite recursion here.